From 0c341f47adc3578cd5f817aa20ee2b7f9ae6b23e Mon Sep 17 00:00:00 2001
From: Phil Elwell <phil@raspberrypi.com>
Date: Wed, 17 Apr 2024 15:57:07 +0100
Subject: [PATCH 1037/1085] dts: bcm2712: cm5: Force the ANT pins to GPIOs

The magic required to make requesting a GPIO automatically change the
pinmux settings is currently absent. Work around that by explicitly
requesting that the ANT pins be in GPIO mode.

Signed-off-by: Phil Elwell <phil@raspberrypi.com>
---
 arch/arm/boot/dts/broadcom/bcm2712-rpi-cm5.dtsi | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

--- a/arch/arm/boot/dts/broadcom/bcm2712-rpi-cm5.dtsi
+++ b/arch/arm/boot/dts/broadcom/bcm2712-rpi-cm5.dtsi
@@ -360,6 +360,9 @@ dpi_16bit_gpio2:        &rp1_dpi_16bit_g
 };
 
 &pinctrl_aon {
+	ant_pins: ant_pins {
+		function = "gpio";
+		pins = "aon_gpio5", "aon_gpio6";
 	};
 
 	/* Slight hack - only one PWM pin (status LED) is usable */
@@ -462,7 +465,7 @@ dpi_16bit_gpio2:        &rp1_dpi_16bit_g
 
 /* SDIO2 drives the WLAN interface */
 &sdio2 {
-	pinctrl-0 = <&sdio2_30_pins>;
+	pinctrl-0 = <&sdio2_30_pins>, <&ant_pins>;
 	pinctrl-names = "default";
 	bus-width = <4>;
 	vmmc-supply = <&wl_on_reg>;
